packages/core/lib/async-every.d.ts

```diff
@@ -0,0 +1,7 @@
+type NodeCallbackType<T = any> = (error?: Error | null, result?: T) => void;
```

Comment:
[TypeScripts looks for sibling type declaration files for relative imports](https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies). Whilst TypeScript will infer types from JS files when `allowJs` is true it's not often sufficient. Without these files many test cases need extra type decoration. In the case of lib files as here I think this is a good intermediate step towards full TypeScript conversion.